File: module_open_projects/src/main/java/com/assess15/module_open_projects/dagger2/demo9/MouseModule.java
package com.assess15.module_open_projects.dagger2.demo9;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;

@Module
public class MouseModule {

    @Provides
    @Named("wired")
    Mouse ProvidesWireMouse() {
        return new WireMouse();
    }

    @Provides
    @Named("wireless")
    Mouse ProvidesWirelessMouse() {
        return new WirelessMouse();
    }
}

File: module_open_projects/src/main/java/com/laychv/module_open_projects/dagger2/demo10/MouseModule.kt
package com.laychv.module_open_projects.dagger2.demo10

import dagger.Module
import dagger.Provides
import javax.inject.Named

@Module
class MouseModule {

    @Provides
    @Named("wired")
    fun providesWireMouse(): Mouse {
        return WireMouse()
    }

    @Provides
    @Named("wireless")
    fun providesWirelessMouse(): Mouse {
        return WirelessMouse()
    }
}
